### PR TITLE
Implement new format of the health report

### PR DESCRIPTION
**Pull request by anjalshireesh**
To ensure compatibility with older versions of `mc`, `minio` server will continue
to return the health info in the current format. `mc` will internally transform/map
this data to the new data structure required to generate the output in the new
format. The new format is as follows:

 - hardware
   - servers
     - cpus
     - meminfo
     - perf
       - net
       - drives
 - software
  - minio
    - info
      - servers
      - drives
    - config
    - procinfos
  - osinfos

The new format also includes a header at the top of the report that provides
information about the version of the report schema, and looks like:

```json
{
    "subnet": {
      "health": {
        "version": "v1"
      }
    }
  }
```

### CODE REVIEW

File: cmd/health_v1.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+		ch.Status = "Error"
+		ch.Error = err.Error()
+		return ch
+	}
+
+	ch.Status = "Success"
+
+	serverAddrs := set.NewStringSet()
+
+	serverCPUs := mapServerCPUs(healthInfo)
```
Comment (Praveenrajmani) [inline cmd/health_v1.go:151]: Suggestion (Ignore if you don't think it makes much difference here) - Can we parallelize the map calls here..?

Comment (Praveenrajmani) [reply cmd/health_v1.go:151]: Something like the following should work If we decide to parallelize. 

```
var serverCPUs map[string][]HwCPUV1
var serverDrives map[string][]HwDriveV1
var serverMems map[string]HwMemV1
..
..
fetchServerCPUFunc := func() {
    serverCPUs = mapServerCPUs(healthInfo)
}
fetchServerDrivesFunc = func() {
    serverDrives = mapServerDrives(healthInfo)
}
..
..
Parallelize(fetchServerCPUFunc, fetchServerDrivesFunc, ..) // Will execute the functions and wait
..
..
// Parallelize parallelizes the function calls
func Parallelize(functions ...func()) {
    var waitGroup sync.WaitGroup
    waitGroup.Add(len(functions))

    defer waitGroup.Wait()

    for _, function := range functions {
        go func(fn func()) {
            defer waitGroup.Done()
             fn()
        }(function)
    }
}
```

**Review by Praveenrajmani — COMMENTED** [review]
Few comments.

**Comment by Praveenrajmani** [comment]
Can we also add an entry in `auto-complete.go` and also update the structure here in `play-obd_20200507160409.json` (Not sure where are we using this data)

File: cmd/admin-subnet-health.go [hunk 2]
```diff
@@ -30,23 +31,21 @@ import (
...
 		Value:  nil,
 		EnvVar: "MC_HEALTH_TEST,MC_OBD_TEST",
 		Hidden: true,
 	},
 	cli.DurationFlag{
 		Name:   "deadline",
-		Usage:  "maximum duration that OBD tests should be allowed to run",
+		Usage:  "maximum duration that health tests should be allowed to run",
 		Value:  3600 * time.Second,
 		EnvVar: "MC_HEALTH_DEADLINE,MC_OBD_DEADLINE",
```
[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/admin-subnet-health.go:50]: The ENV var supports both the previous (`MC_OBD_DEADLINE`) and new (`MC_HEALTH_DEADLINE`) key

**Comment by anjalshireesh** [comment]
> Can we also add an entry in `auto-complete.go` and also update the structure here in `play-obd_20200507160409.json` (Not sure where are we using this data)

The json file should not be part of the PR! Will remove it.

File: cmd/admin-subnet-health.go [hunk 3]
```diff
@@ -55,9 +54,9 @@ var adminOBDFlags = []cli.Flag{
 var adminSubnetHealthCmd = cli.Command{
```
[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/admin-subnet-health.go:54]: `health` is already a subcommand of `subnet`. Will add the `CommandNotFound` wrapper

**Comment by anjalshireesh** [comment]
> Can we also add an entry in `auto-complete.go` and ~also update the structure here in `play-obd_20200507160409.json` (Not sure where are we using this data)~ (Ignore)

Following entry already exists. Anything else required?

`"/admin/subnet/health": aliasCompleter,`

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by Praveenrajmani — APPROVED** [review]
LGTM

**Pull request merged**